config/sets/symfony/symfony64.php

```diff
@@ -12,6 +12,7 @@
         [
             'Symfony\Component\HttpKernel\UriSigner' => 'Symfony\Component\HttpFoundation\UriSigner',
             'Symfony\Component\HttpKernel\Debug\FileLinkFormatter' => 'Symfony\Component\ErrorHandler\ErrorRenderer\FileLinkFormatter',
+            'Symfony\Component\Routing\Annotation\Route' => 'Symfony\Component\Routing\Attribute\Route',
```

Comment:
if `AnnotationToAttributeRector` is not called early, it may cause unexpected result, since it will still annotation, while renamed.

I think `RenameClassRector` need to be placed right **after** `AnnotationToAttributeRector` in the config annotation to attribute there

https://github.com/rectorphp/rector-symfony/blob/c9d78b386a7658e3f9ad71ad055a38eca436ee69/config/sets/symfony/annotations-to-attributes.php#L14

Reply:
On AnnotationToAtrributeRector, the

`new AnnotationToAttribute('Symfony\Component\Routing\Annotation\Route' , 'Symfony\Component\Routing\Attribute\Route')`

should auto rename

Reply:
I'm in the train myself, not the best place to work. ^^ Thanks to point me to the right way. Do you have recommandation to learn rector ecosystem ?

Reply:
@TomasVotruba write a book about rector

https://tomasvotruba.com/book-detail/rector-the-power-of-automated-refactoring?slug=rector-the-power-of-automated-refactoring

Reply:
I'll ask my company to buy it :)